frontend/src/components/CheckData.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useMemo } from "react";
+import Button from "./Button";
+import Table from "./Table";
+
+interface Props {
+  selectedHeaders: Set<string>;
+  hiddenColumns: Set<string>;
+  register: Function;
+  setSelectedHeaders: Function;
+  setHiddenColumns: Function;
+  backStep: Function;
+  advanceStep: Function;
+  onCancel: Function;
+  data: Array<any>;
+}
+
+function CheckData(props: Props) {
+  const handleSelectedHeadersChange = (
+    event: React.FormEvent<HTMLInputElement>
+  ) => {
+    const target = event.target as HTMLInputElement;
+    const newSelectedHeaders = new Set(props.selectedHeaders);
+    if (target.checked) {
+      newSelectedHeaders.add(target.name);
+    } else {
+      newSelectedHeaders.delete(target.name);
+    }
+    props.setSelectedHeaders(newSelectedHeaders);
+  };
+
+  const handleHideFromVisualizationChange = (
+    event: React.FormEvent<HTMLInputElement>
+  ) => {
+    const target = event.target as HTMLInputElement;
+    const newHiddenColumns = new Set(props.hiddenColumns);
+    if (target.checked) {
+      for (const selectedHeader of Array.from(props.selectedHeaders)) {
+        newHiddenColumns.add(selectedHeader);
+      }
+    } else {
+      for (const selectedHeader of Array.from(props.selectedHeaders)) {
+        newHiddenColumns.delete(selectedHeader);
+      }
+    }
+    props.setHiddenColumns(newHiddenColumns);
+  };
+
+  const checkDataTableRows = useMemo(() => props.data || [], [props.data]);
+  const checkDataTableColumns = useMemo(
+    () =>
+      (props.data.length
+        ? (Object.keys(props.data[0]) as Array<string>)
+        : []
+      ).map((header, i) => {
+        return {
+          Header: () => (
+            <span className="text-center usa-checkbox margin-bottom-1">
+              <input
+                className="usa-checkbox__input"
+                id={`checkbox-header-${i}`}
+                type="checkbox"
+                name={header}
+                defaultChecked={props.selectedHeaders.has(header)}
+                onChange={handleSelectedHeadersChange}
+                ref={props.register()}
+              />
+              <label
+                className="usa-checkbox__label"
+                htmlFor={`checkbox-header-${i}`}
+              ></label>
```

Comment:
The content of this label is empty, I suggest you should add something so that screen readers can understand what his checkbox is for. For example: 

```jsx
<label className="sr-only"> Select column {i} </label>
```

Reply:
I tried it but it is not working.